cli/dcoscli/node/main.py

```diff
@@ -473,24 +477,177 @@ def _log(follow, lines, leader, slave):
     :type leader: bool
     :param slave: the slave ID to print
     :type slave: str | None
+    :param list_components: list DC/OS components for a given node
+    :type list_components: bool
+    :param component: DC/OS component name
+    :type component: string
+    :param filters: a list of filters ["key:value", ...]
+    :type filters: list
     :returns: process return code
     :rtype: int
     """
 
     if not (leader or slave):
-        raise DCOSException('You must choose one of --leader or --mesos-id.')
+        raise DCOSException(
+            'You must choose one of --leader or --mesos-id.')
+
+    # validate mesos agent node id
+    slave_ip = _get_slave_ip(slave)
+
+    if list_components:
+        _list_components(leader, slave_ip)
+        return 0
 
     if lines is None:
         lines = 10
+
     lines = util.parse_int(lines)
 
+    try:
+        _dcos_log(follow, lines, leader, slave, component, filters)
+        return 0
+    except DCOSException as e:
+        emitter.publish(DefaultError(e))
+        emitter.publish(DefaultError('Falling back to files API...'))
+    except (DCOSAuthenticationException,
```

Comment:
`DCOSAuthenticationException` and `DCOSAuthorizationException` should be caught before `DCOSException` because they also are a `DCOSException`